src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultRazorLanguageServerCustomMessageTarget.cs

```diff
@@ -724,6 +734,11 @@ public override async Task<VSInternalWrapWithTagResponse> RazorWrapWithTagAsync(
             return null;
         }
 
+        if (htmlRanges is null || csharpRanges is null)
```

Comment:
Without this change when html or c# returns null we throw down below. We could alternatively give them null-forgiveness, but this seemed closer to what we do in SemanticTokens.

Reply:
The problem is that if we return a non-null value as the result of the `ProvideFoldingRangesAsync` call VS thinks it's got a complete view of the folding ranges and doesn't need to re-query (because it doesn't poll) until there's been an edit. So we'll get no ranges for whichever language wasn't ready yet until the document is edited.

I didn't want to change other spots because I don't know it to be an issue there and you're correct that otherwise it's preferable to return at least some results. I've added a comment to that effect.

Reply:
I forgot folding ranges were special like this.. sorry, I now remember that you've explained this to me before :)